dev/NavigationView/NavigationView.cpp

```diff
@@ -1923,7 +1933,17 @@ void NavigationView::UpdateSingleSelectionFollowsFocusTemplateSetting()
 void NavigationView::OnSelectedItemPropertyChanged(winrt::DependencyPropertyChangedEventArgs const& args)
 {
     auto newItem = args.NewValue();
-    ChangeSelection(args.OldValue(), newItem);
+    auto oldItem = args.OldValue();
+    ChangeSelection(oldItem, newItem);
+
+    // Animate to be sure the selected item is visually higlighted!
+    // See #1395 for additional context
+    if (oldItem != newItem)
```

Comment:
Don't do it here. Some feature like NavigationRecommendedTransitionDirection, SelectionSuppressed and topnav animation may be broken which is implemented in ChangeSelection.

Reply:
Hmm I see. So should we set `m_shouldRaiseInvokeItemInSelectionChange` to false when before we call `ChangeSelection` in `OnSelectedItemPropertyChanged`? Or would it be better to move lines 1943/1944 inside `CreateAndHookEventsToSettings` ?

Reply:
I don't have good suggestion on it. It's possible that add a flag like 'Hi, I'm switching from topnav to leftnav, please don't raise selection change event for settings'.
Most likely it's not a `easy` change because we don't have too much automation on this part and you have to manually verify all kind of scenarios.
I would like @YuliKl to re-evaluate the impact of the bug.

Reply:
I admit that I'm not quite following the code intricacies within NavigationView and dont' have a good grasp on what's complicating this fix. I do know there's a real problem outlined in #148 that was encountered by a customer (although at this point, I can't locate any record of who originally found this bug).

Reply:
This seems to be stalling out. @ojhad @licanhua @YuliKl @chingucoding any idea what the next steps are?

Reply:
I think the least we should do, is to ensure that the selected item that gets passed to the event-handler/event-args is not null. 

However not raising the event in that case seems to be quite difficult, as we apparently break animations with that. The question is how common it is for developers to change the orientation of the NavigationView during runtime. If this is an edge case that is not very common, we should maybe not try to prevent the raising of that event. After all, catching that "bug" is something that is definitely achievable.